actix-http/src/header/shared/quality_item.rs

```diff
@@ -21,9 +28,51 @@ use self::internal::IntoQuality;
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Quality(u16);
 
+impl Quality {
+    /// # Panics
+    /// Panics in debug mode when value is not in the range 0.0 <= n <= 1.0.
+    fn from_f32(value: f32) -> Self {
+        // Check that `value` is within range should be done before calling this method.
+        // Just in case, this debug_assert should catch if we were forgetful.
+        debug_assert!(
```

Comment:
Only because it's private and checks for the range *should* be done before using this somewhere in the lib code (as it is in the header parsing code). So it's wasteful to check the range twice in release mode.